File: src/main/java/net/onelitefeather/pandorascluster/land/player/LandMember.java
package net.onelitefeather.pandorascluster.land.player;

import jakarta.persistence.*;
import net.onelitefeather.pandorascluster.enums.LandRole;
import net.onelitefeather.pandorascluster.land.Land;
import org.jetbrains.annotations.NotNull;

@Entity
public class LandMember {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @OneToOne
    private LandPlayer member;

    @Enumerated(EnumType.STRING)
    private LandRole role;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "land_id")
    private Land land;

    public Land getLand() {
        return land;
    }

    public void setLand(Land land) {
        this.land = land;
    }

    public LandMember() {
    }

    public LandMember(long id, @NotNull LandPlayer member, @NotNull LandRole role) {
        this.id = id;
        this.member = member;
        this.role = role;
    }

    public LandMember(LandPlayer member, LandRole role, Land land) {
        this.member = member;
        this.role = role;
        this.land = land;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @NotNull
    public LandPlayer getMember() {
        return member;
    }

    public void setMember(@NotNull LandPlayer member) {
        this.member = member;
    }

    @NotNull
    public LandRole getRole() {
        return role;
    }

    public void setRole(@NotNull LandRole role) {
        this.role = role;
    }
}

File: src/main/kotlin/net/onelitefeather/pandorascluster/land/player/LandMember.kt
package net.onelitefeather.pandorascluster.land.player

import jakarta.persistence.*
import net.onelitefeather.pandorascluster.enums.LandRole
import net.onelitefeather.pandorascluster.land.Land
import net.onelitefeather.pandorascluster.util.Constants
import org.hibernate.Hibernate

@Entity
data class LandMember(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @OneToOne
    val member: LandPlayer? = null,

    @Enumerated(EnumType.STRING)
    val role: LandRole = LandRole.VISITOR,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "land_id")
    val land: Land = Constants.DUMMY_LAND

) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as LandMember

        return id != null && id == other.id
    }

    override fun hashCode(): Int = javaClass.hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , member = $member , role = $role )"
    }
}